compiler/rustc_middle/src/ty/visit.rs

```diff
@@ -542,6 +547,20 @@ impl<'tcx> TypeVisitor<'tcx> for HasEscapingVarsVisitor {
         }
     }
 
+    fn visit_effect(&mut self, e: ty::Effect<'tcx>) -> ControlFlow<Self::BreakTy> {
+        // we don't have a `visit_infer_effect` callback, so we have to
+        // hook in here to catch this case (annoying...), but
+        // otherwise we do want to remember to visit the rest of the
+        // effect, as it has types/regions embedded in a lot of other
+        // places.
```

Comment:
i don't understand that comment :thinking:

Reply:
I think I copy pasted it from constants without looking at the details